src/metrics/misc.ts

```diff
@@ -242,6 +250,37 @@ export class MetadataMetric implements Metric {
       };
     }
 
+    const chess = new Chess();
+
+    // identify game endings from the game
+    for (const { move } of game) {
+      chess._makeMove({
+        from: move.fromIndex,
+        to: move.toIndex,
+        promotion: move.promotion,
+        color: move.color,
+        piece: move.piece,
+        uas: move.uas,
+        flags: parseInt(move.flags, 10)
+      });
+
+      if (chess.isThreefoldRepetition()) {
```

Comment:
realizing the if statements below should prob be outside of the loop? not sure actually